CONTRIBUTING.md

```diff
@@ -21,6 +21,46 @@ The files in this project are formatted by [Prettier](https://prettier.io/) and
 npm run eslint
 ```
 
+### When to use HTMLElement vs Virtual Node
+
+Axe-core uses an internal copy of the HTML page for most things internal to axe-core (called the Virtual Tree). Each HTML element on the page has an equivalent [Virtual Node](./lib/core/base/virtual-node) element, which allows us to cache or normalize information about the HTML element without mutating the actual DOM node.
+
+Typically we use the Virtual Node when possible, but understand that's not always possible (such as accessing DOM only APIs). Furthermore, any function within the [utils directory](./lib/core/utils/) should not use Virtual Nodes. The reason for this is that using Virtual Nodes requires that axe first be [setup](./lib/core/public/setup.js) and create the Virtual Tree. For the most part, util functions are functions that can be run when no Virtual Tree exists.
```

Comment:
> (such as accessing DOM only APIs)

Is there an example you could link to for this? Understanding this tripped me up a little.